src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/CodeReaderWriter/ProjectModifier.cs

```diff
@@ -0,0 +1,611 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Text;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.DotNet.MSIdentity.AuthenticationParameters;
+using Microsoft.DotNet.MSIdentity.Properties;
+using Microsoft.DotNet.MSIdentity.Tool;
+
+namespace Microsoft.DotNet.MSIdentity.CodeReaderWriter
+{
+    internal class ProjectModifier
+    {
+        private List<CodeModifierConfig> _codeModifierConfigs;
+        private readonly ProvisioningToolOptions _toolOptions;
+        private readonly ApplicationParameters _appParameters;
+        private readonly IConsoleLogger _consoleLogger;
+        public SyntaxTrivia SemiColonTrivia
+        {
+            get
+            {
+                return SyntaxFactory.Trivia(SyntaxFactory.SkippedTokensTrivia()
+                    .WithTokens(SyntaxFactory.TokenList(SyntaxFactory.Token(SyntaxKind.SemicolonToken))));
+            }
+        }
+        public ProjectModifier(ApplicationParameters applicationParameters, ProvisioningToolOptions toolOptions, IConsoleLogger consoleLogger)
+        {
+            _toolOptions = toolOptions ?? throw new ArgumentNullException(nameof(toolOptions));
+            _appParameters = applicationParameters ?? throw new ArgumentNullException(nameof(applicationParameters));
+            _consoleLogger = consoleLogger ?? throw new ArgumentNullException(nameof(consoleLogger));
+            _codeModifierConfigs = new List<CodeModifierConfig>();
+        }
+
+        internal IDictionary<string, string>? VerfiyParameters(string[]? parametersToCheck, List<ParameterSyntax> foundParameters)
+        {
+            IDictionary<string, string> parametersWithNames = new Dictionary<string, string>();
+            if (foundParameters.Any())
+            {
+                var pars = foundParameters.ToList();
+                if (parametersToCheck != null)
+                {
+                    foreach (var parameter in parametersToCheck)
+                    {
+                        var verifiedParams = pars.Where(p => p.Type != null && p.Type.ToFullString().Trim().Equals(parameter.Trim(), StringComparison.OrdinalIgnoreCase));
+                        if (!verifiedParams.Any())
+                        {
+                            return null;
+                        }
+                        else
+                        {
+                            parametersWithNames.Add(parameter, verifiedParams.First().Identifier.ValueText);
+                        }
+                    }
+                }
+            }
```

Comment:
you have 3 nesting levels here .. As a little example related to a huge method below, you could validate parameters in the first line and return 
after that yu owould have on ly one nesting - easier to read and maintain.

Reply:
reduced significantly.